src/lib/routing/router.ts

```diff
@@ -20,21 +20,54 @@ const defaultErrorHandler = (auto405) => async (err: Error, req: EReq, res: ERes
   res.withStatus(500).json({ error: err.message });
 }
 
+/**
+ * Handles preflight requests from trusted origins configured by the user when initializing a router.
+ * Note that the wildcard value "*" will fail if the request is sent with credentials (see: https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Access-Control-Allow-Origin#directives)
+ * @param autoCorsPreflight the object containing CORS preflight option of trustedOrigins, an array of trusted origins.
+ * @returns 200 if the preflight request succeeds, 403 if it fails
+ */
+const preflightHandler = (autoCorsPreflight: AutoCorsPreflightOptions) => async (req: EReq, res: ERes) => {
+  if (autoCorsPreflight.trustedOrigins.length === 0) {
+    return res.sendStatus(403);
+  }
+  let originHeaderValue: string | null = null;
+  if (autoCorsPreflight.trustedOrigins.length === 1 && autoCorsPreflight.trustedOrigins[0] === '*') {
+    originHeaderValue = '*';
+  } else {
+    const origin = req.headers.get("origin").toLowerCase();
+    if (autoCorsPreflight.trustedOrigins.some((trustedOrigin) => trustedOrigin.toLowerCase() === origin)) {
+      originHeaderValue = origin;
+    }
+  }
+  if (originHeaderValue == null) {
+    return res.sendStatus(403);
+  }
+  res.set({
+    'access-control-allow-origin': originHeaderValue,
+    'access-control-allow-methods': req.headers.get("access-control-request-method"),
+    'access-control-allow-headers': req.headers.get("access-control-request-headers"),
```

Comment:
This will still output "null" if there is no `access-control-request-headers` header on the request, which means we are telling the browser it's OK to send a request with a header called "null".  You need to only include this response header if the request header is there:

```
if (req.headers.has("access-control-request-method")) {
  res.headers.set("access-control-allow-methods", req.headers.get("access-control-request-method"));
}
```